operator/src/config.rs

```diff
@@ -78,188 +67,213 @@ pub fn create_config_map_name(pod_name: &str) -> String {
     format!("{}-config", pod_name)
 }
 
-/// Create all required config maps and respective config map data.
+/// Filter all existing pods for master node type and retrieve the role group config
+/// Extract the nodeName from the pod and the specified port from the config to create
+/// the master urls for each pod.
 ///
 /// # Arguments
-/// * `resource` - SparkCluster
-/// * `config` - The custom resource config
-/// * `cm_name` - The desired config map name
+/// * `pods`             - Slice of all existing pods
+/// * `validated_config` - The precalculated role and role group configuration with properties
+///                        split by PropertyNameKind
+pub fn get_master_urls(
+    pods: &[Pod],
+    validated_config: &ValidatedRoleConfigByPropertyKind,
+) -> OperatorResult<Vec<String>> {
+    let mut master_urls = Vec::new();
+
+    for pod in pods {
+        if let (Some(role), Some(group)) = (
+            pod.metadata.labels.get(APP_COMPONENT_LABEL),
+            pod.metadata.labels.get(APP_ROLE_GROUP_LABEL),
+        ) {
+            if role != &SparkRole::Master.to_string() {
+                continue;
+            }
+
+            let validated_config_for_role_and_group =
+                config_for_role_and_group(role, group, validated_config)?;
+
+            let mut port = "7077";
```

Comment:
Why do we need to hardcode one here? Wouldn't we get the default from our config?